src/net/socket.rs

```diff
@@ -0,0 +1,222 @@
+use crate::config::NetworkConfig;
+use crate::error::{NetworkError, NetworkErrorKind, NetworkResult};
+use crate::net::link_conditioner::LinkConditioner;
+use crate::net::{connection::ActiveConnections, events::SocketEvent};
+use crate::packet::Packet;
+use log::error;
+use mio::{Evented, Events, Poll, PollOpt, Ready, Token};
+use std::io;
+use std::mem;
+use std::net::{SocketAddr, ToSocketAddrs};
+use std::sync::{mpsc, Arc};
+
+const SOCKET: Token = Token(0);
+
+/// A reliable UDP socket implementation with configurable reliability and ordering guarantees.
+pub struct LaminarSocket {
```

Comment:
As discussed on discord this should be renamed to `Socket` with also `NetworkConfig` -> `Config`. `Socket` can still live under net but can be reexported at the root of the project so that people can access them as `laminar::{Socket, Config}`.